### PR TITLE
Add test to app

### PR DESCRIPTION
**Pull request by jaamanu**
In this PR
Made sure that the N+1 problem is solved when fetching all posts and their comments for a user.

 Analyzed the logs as showed in this [article](https://dev.to/junko911/rails-n-1-queries-and-eager-loading-10eh)
 Included screenshots of the app's logs before and after the fix in the PR description.

Screenshot1
Screenshot2

Integration specs
Use Capybara to write integration tests for each view in your project.

User index page:

I can see the username of all other users.
I can see the profile picture for each user.
I can see the number of posts each user has written.
When I click on a user, I am redirected to that user's show page.
user show page:

I can see the user's profile picture.
I can see the user's username.
I can see the number of posts the user has written.
I can see the user's bio.
I can see the user's first 3 posts.
I can see a button that lets me view all of a user's posts.
When I click a user's post, it redirects me to that post's show page.
When I click to see all posts, it redirects me to the user's post's index page.
User post index page:

I can see the user's profile picture.
I can see the user's username.
I can see the number of posts the user has written.
I can see a post's title.
I can see some of the post's body.
I can see the first comments on a post.
I can see how many comments a post has.
I can see how many likes a post has.
I can see a section for pagination if there are more posts than fit on the view.
When I click on a post, it redirects me to that post's show page.
Post show page:

I can see the post's title.
I can see who wrote the post.
I can see how many comments it has.
I can see how many likes it has.
I can see the post body.
I can see the username of each commentor.
I can see the comment each commentor left.

### CODE REVIEW

File: spec/integration/post/index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    it 'Can see a pagination button if there are more posts than fit' do
+      expect(page).to have_content('Pagination')
+    end
+
+    it 'Should have a post titkle' do
+      expect("#{page.current_url}/#{@post.id}").to match user_post_path(@user.id, @post.id)
+    end
+  end
+end
```
Comment (franclobo) [inline spec/integration/post/index_spec.rb:52]: - [ ] Kindly review your integration test, all the test should pass to approve the milestone 👇🏽 

image

File: spec/integration/user/show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      expect(page).to have_content('See all posts')
+    end
+    it 'should return each user profile picture' do
+      expect("#{page.current_url}/posts/#{@post.id}").to match user_post_path(@user, @post)
+    end
+    it 'should return each user profile picture' do
+      expect("#{page.current_url}/posts").to match user_posts_path(@user)
+    end
+  end
+end
```
Comment (franclobo) [inline spec/integration/user/show_spec.rb:35]: - [ ] Kindly review your integration test, all the test should pass to approve the milestone 👇🏽 

image

File: spec/integration/user/index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    it 'should return each user profile picture' do
+      @users.each do |pic|
+        expect(page).to have_content(pic.post_counter)
+      end
+    end
+    it 'should return each user profile picture' do
+      expect("#{page.current_url}/#{@user.id}").to match users_path(@user.id)
+    end
+  end
+end
```
Comment (franclobo) [inline spec/integration/user/index_spec.rb:29]: - [ ] Kindly review your integration test, all the test should pass to approve the milestone 👇🏽 

image

File: spec/integration/post/show_spec.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      end
+    end
+
+    it '' do
+      @post.comments.each do |comment|
+        expect(page).to have_content(comment.text)
+      end
+    end
+  end
+end
```
Comment (franclobo) [inline spec/integration/post/show_spec.rb]: - [ ] Kindly review your integration test, all the test should pass to approve the milestone 👇🏽 

image

File: spec/integration/post/show_spec.rb [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+      @user = User.create(name: 'james',
+                          photo: 'https://img.freepik.com/free-psd/save-world-reusable-grocery-bag-mockup_53876-98827.jpg?', bio: 'james')
+      @post = Post.create(text: 'ken', title: 'ken', author: @user)
+      Comment.create(post: @post, author: @user, text: 'Amazing post Tom')
+      Comment.create(post: @post, author: @user, text: 'Awesome post Tom')
+      Comment.create(post: @post, author: @user, text: 'Great post Tom')
+      visit user_post_path(@user.id, @post)
+    end
+
+    it 'Should have a post titkle' do
```
Comment (franclobo) [inline spec/integration/post/show_spec.rb]: - [ ] May you mean title? Please 🙏🏽 write the correct word for professional documentation.

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @jaamanu, 

# ♻️ Required Changes ♻️

so-close-jim-halpert

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Capybara is used 💯 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/integration/post/show_spec.rb (hunk 4 above)
Comment (jaamanu) [reply spec/integration/post/show_spec.rb]: Hi@franclobo
image

All testing all passing on my local drive as you can see the attached picture.

File: spec/integration/post/show_spec.rb (hunk 5 above)
Comment (jaamanu) [reply spec/integration/post/show_spec.rb]: i have modified the comments as well

File: spec/integration/user/show_spec.rb (hunk 2 above)
Comment (jaamanu) [reply spec/integration/user/show_spec.rb:35]: Hi@franclobo
image

All testing all passing on my local drive as you can see the attached picture.

File: spec/integration/post/index_spec.rb (hunk 1 above)
Comment (jaamanu) [reply spec/integration/post/index_spec.rb:52]: Hi@franclobo
image

All testing all passing on my local drive as you can see the attached picture.

**Review by Nemwel-Boniface — APPROVED** [review]
## Hello @jaamanu        :wave: ,
I am Nemwel your reviewer for this PR :man_beard:  

Thank you for meeting all the project requirements good job! :partying_face: 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! :partying_face: 🎉 



## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️ 
Met all the milestone requirements ✔️ 

## [Optional] suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**